src/core/Typography/Text/Text.tsx

```diff
@@ -53,9 +53,7 @@ export type TextProps<
  * <Text isMuted>Some muted text.</Text>
  * <Text isSkeleton>Skeleton text</Text>
  */
-export const Text = <T extends React.ElementType | React.ComponentType = 'div'>(
-  props: TextProps<T>,
-) => {
+export const Text: TextComponent = React.forwardRef((props, ref) => {
```

Comment:
We can't have `as` without `forwardRef`?

Reply:
I think we could, but it's better to forward the ref wherever possible. Our tooltip, for example, only works on elements that forward their ref